src/pages/Portfolio/ImpactDashboard/LoanCommentModal.vue

```diff
@@ -0,0 +1,128 @@
+<template>
+	<kv-lightbox
+		:visible="loan.visible"
+		:title="modalTitle"
+		@lightbox-closed="closeModal"
+	>
+		<div class="tw-text-center">
+			<div class="tw-relative">
+				<textarea
+					class="tw-w-full tw-border tw-border-secondary tw-rounded-sm tw-h-7 tw-p-1 comment-input"
+					v-model="userComment"
+					:placeholder="commentPlaceholder"
+				>
+			</textarea>
+				<kv-material-icon
+					class="tw-w-2.5 tw-h-2.5 tw-absolute tw-bottom-2 tw-right-0"
+					:icon="mdiPencilOutline"
+				/>
+			</div>
+			<small
+				class="tw-text-left tw-block"
+			>Please follow our
+				<a
+					href="/kiva-community-guidelines" target="_blank"
+					class="tw-text-action hover:tw-text-action-highlight"
+				>
+					community guidelines
+				</a>
+			</small>
+		</div>
+		<kv-button
+			class="tw-mt-2"
+			variant="primary"
+			aria-label="Comment"
+			:state="buttonState"
+			@click="submitComment"
+		>
+			Save
+		</kv-button>
+	</kv-lightbox>
+</template>
+
+<script>
+import logFormatter from '@/util/logFormatter';
+import { mdiPencilOutline } from '@mdi/js';
+import loanAddComment from '@/graphql/mutation/loanAddComment.graphql';
+import KvLightbox from '~/@kiva/kv-components/vue/KvLightbox';
+import KvMaterialIcon from '~/@kiva/kv-components/vue/KvMaterialIcon';
+import KvButton from '~/@kiva/kv-components/vue/KvButton';
+
+export default {
+	name: 'LoanCommentModal',
+	inject: ['apollo', 'cookieStore'],
+	components: {
+		KvLightbox,
+		KvMaterialIcon,
+		KvButton,
+	},
+	props: {
+		loan: {
+			type: Object,
+			default: () => {}
+		}
+	},
+	data() {
+		return {
+			mdiPencilOutline,
+			userComment: '',
+			loading: false,
+		};
+	},
+	computed: {
+		buttonState() {
+			if (this.loading) return 'loading';
+			if (!this.userComment) return 'disabled';
+			return '';
+		},
+		modalTitle() {
+			return `Tell others why you love this loan to ${this.loan.borrowerName}`;
+		},
+		commentPlaceholder() {
+			return `Tell others why this loan to ${this.loan.borrowerName} is great!`;
+		}
+	},
+	methods: {
+		submitComment() {
+			this.loading = true;
+			this.apollo.mutate({
+				mutation: loanAddComment,
+				variables: {
+					id: this.loan.loanId,
+					body: this.userComment
+				}
+			}).then(({ data }) => {
+				// comment was added successfully
+				if (data.loan.addComment) {
+					this.closeModal();
+					this.$kvTrackEvent('portfolio', 'click', 'Leave a loan comment', this.loan.borrowerName, this.loan.loanId); // eslint-disable-line max-len
+					this.$showTipMsg(`Thank you for helping ${this.loan.borrowerName}!`, 'confirmation', true);
+					this.$router.push(`/lend/${this.loan.loanId}`);
+				} else {
+					throw new Error('Comment not added');
+				}
+			}).catch(e => {
+				logFormatter(e, 'error');
+				this.$showTipMsg('There was a problem commenting on this loan', 'error');
```

Comment:
Since we don't close the modal on error, will the tip message work when shown with the open modal? Do we need to also close the modal on error?

Reply:
Thanks for this catch!
A possible error here is b/c any API issue, so, I don't think closing the modal is a good UX.
I think we can add the error message below the `Save` button, adding a `try again` copy, in this way the user don't need to open de modal and write the comment again.